101-automation-configuration/azuredeploy.json

```diff
@@ -0,0 +1,301 @@
+      "defaultValue": "10.0.0.0/24"
+    },
+    "nicIPAddress": {
+      "type": "string",
+      "metadata": {
+        "description": "The IP address of the new AD VM"
+      },
+      "defaultValue": "10.0.0.4"
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Administrator of the new VM and Domain"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password for the Administrator account of the new VM and Domain"
+      }
+    },
+    "safeModePassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password for the safe mode restore feature in the new domain"
+      }
+    },
+    "VMSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The size of the VM Created"
+      },
+      "defaultValue": "Standard_DS2_V2"
+    },
+    "domainName": {
+      "type": "string",
+      "metadata": {
+        "description": "The full qualified domain name to be created"
+      },
+      "defaultValue": "contoso.local"
+    }
+  },
+  "variables": {
+    "automationAccountName": "[concat(parameters('prefix'), 'configuration')]",
+    "publicIPAddressName": "publicIP",
+    "publicIPAddressType": "Dynamic",
+    "virtualNetworkSubnetName": "Subnet",
+    "virtualNetworkSubnets": [
+      {
+        "name": "[variables('virtualNetworkSubnetName')]",
+        "properties": {
+          "addressPrefix": "[parameters('virtualNetworkSubnet')]"
+        }
+      }
+    ],
+    "VMName": "server",
+    "DataDisk": "DataDisk",
+    "DataDiskSize": 1000,
+    "nicName": "nic",
+    "virtual networkID": "[resourceId('Microsoft.Network/virtualNetworks', parameters('virtualNetworkName'))]",
+    "nicSubnetRef": "[concat(variables('virtual networkID'),'/subnets/',variables('virtualNetworkSubnetName'))]",
+    "PublicIpResourceId": "[resourceId(concat('Microsoft.Network','/','publicIPAddresses'),variables('publicIPAddressName'))]",
+    "NicId": "[resourceId('Microsoft.Network/networkInterfaces',variables('nicName'))]",
+    "IPConfigID": "[concat(variables('NicId'),'/ipConfigurations/ipconfig')]",
+    "nestedTemplateBaseURL": "https://raw.githubusercontent.com/mgreenegit/azure-quickstart-templates/101-automation-configuration/101-automation-configuration/nested/",
+    "provisionConfigurationURL": "[uri(variables('nestedTemplateBaseURL'), 'provisionConfiguration.json')]",
+    "provisionStorageURL": "[uri(variables('nestedTemplateBaseURL'), 'provisionStorage.json')]",    
+    "provisionNetworkURL": "[uri(variables('nestedTemplateBaseURL'), 'provisionNetwork.json')]",
+    "provisionServerUrl": "[uri(variables('nestedTemplateBaseURL'), 'provisionServer.json')]",
+    "provisionDNSUrl": "[uri(variables('nestedTemplateBaseURL'), 'provisionDNS.json')]",
+    "storageAccountNamePrefix": "[toLower(concat(parameters('prefix'), substring(resourceGroup().id, 7, 6)))]"
```

Comment:
What's the intent with substring here?  this will pull 'iption' from every deployment - seems like we could just use uniquestring()

Reply:
fixed, thank you